### ISSUE TITLE
Use supported build tools from chipper

### ISSUE THREAD

**Issue by jbphet** [phetsims/rosetta#444]
While working with @zepumph today on https://github.com/phetsims/rosetta/issues/311, he suggested that Rosetta use supported build tools that are available through chipper (and I think perhaps perennial-alias) instead of the piecemeal dependencies that it has right now.  Rosetta has a small set of dependencies on chipper and perennial, but as of now it is not using these dependencies the same way that other NodeJS-based repos are, and it would be good to make things more consistent.

@zepumph said he'd start on this and then potentially assign the issue back to me.

**Comment by zepumph** [comment]
Basically this patch could be called "depend on chipper/perennial-alias instead of perennial". It is sorta blocked by the `publicConfig` issue. I'm not sure how is best to get around that. Most likely we want config in json, and have the js import that file. Then the json is ignored, but the `js` is always checked in. Anywho, that is probably a separate issue (but note my rename in this issue of `tsconfig.json`)


<details>

```diff
Subject: [PATCH] Move MipmapElement to chipper, see https://github.com/phetsims/chipper/issues/1523
---
Index: scripts/js/get-count-of-new-translations.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/js/get-count-of-new-translations.js b/scripts/js/get-count-of-new-translations.js
--- a/scripts/js/get-count-of-new-translations.js	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/scripts/js/get-count-of-new-translations.js	(date 1734467022383)
@@ -49,7 +49,7 @@
 let activeSimList;
 try {
   const initialDirectory = process.cwd();
-  activeSimList = fs.readFileSync( '../../../perennial/data/active-sims', 'utf-8' );
+  activeSimList = fs.readFileSync( '../../../perennial-alias/data/active-sims', 'utf-8' );
   process.chdir( initialDirectory );
 }
 catch( e ) {
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/package.json	(date 1734467022390)
@@ -11,10 +11,10 @@
   "scripts": {
     "lint": "grunt lint",
     "type-check": "grunt type-check",
-    "start-node": "bash ../perennial/bin/sage run src/server/app.ts",
-    "dev-node": "nodemon --exec \"bash ../perennial/bin/sage run src/server/app.ts",
-    "debug": "npm run make-public-config && npm run build --workspace=client && bash ../perennial/bin/sage run --inspect src/server/app.ts",
-    "make-public-config": "bash ../perennial/bin/sage run ./scripts/js/make-public-config-file.mjs",
+    "start-node": "bash ../perennial-alias/bin/sage run src/server/app.ts",
+    "dev-node": "nodemon --exec \"bash ../perennial-alias/bin/sage run src/server/app.ts",
+    "debug": "npm run make-public-config && npm run build --workspace=client && bash ../perennial-alias/bin/sage run --inspect src/server/app.ts",
+    "make-public-config": "bash ../perennial-alias/bin/sage run ./scripts/js/make-public-config-file.mjs",
     "dev": "npm run make-public-config && npm run lint && npm run type-check && concurrently \"npm run dev --workspace=client\" \"npm run dev-node\"",
     "start": "npm run make-public-config && npm run lint && npm run type-check && npm run build --workspace=client && npm run start-node",
     "prod": "npm run make-public-config && npm run build --workspace=client && npm run start-node"
Index: tsconfig-backup.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tsconfig-backup.json b/tsconfig.json
rename from tsconfig-backup.json
rename to tsconfig.json
--- a/tsconfig-backup.json	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/tsconfig.json	(date 1734467022380)
@@ -3,14 +3,14 @@
   "include": [
     "src/common/**/*",
     "src/server/**/*",
-    "../perennial/js/common/simPhetioMetadata.js"
+    "../perennial-alias/js/common/simPhetioMetadata.js"
   ],
   "exclude": [
-    "**/node_modules/**/*",
+    "**/node_modules/**/*"
   ],
   "references": [
     {
-      "path": "../perennial/js/browser-and-node/tsconfig.json"
+      "path": "../chipper/tsconfig/buildtools/tsconfig.json"
     }
   ]
 }
\ No newline at end of file
Index: grunt/tasks/update-rosetta.cjs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/grunt/tasks/update-rosetta.cjs b/grunt/tasks/update-rosetta.cjs
--- a/grunt/tasks/update-rosetta.cjs	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/grunt/tasks/update-rosetta.cjs	(date 1734467022387)
@@ -12,10 +12,10 @@
  */
 'use strict';
 
-const cloneRepo = require( '../../../perennial/js/common/cloneRepo.js' );
+const cloneRepo = require( '../../../perennial-alias/js/common/cloneRepo.js' );
 const fs = require( 'fs' );
-const gitPull = require( '../../../perennial/js/common/gitPull.js' );
-const npmUpdate = require( '../../../perennial/js/common/npmUpdate.js' );
+const gitPull = require( '../../../perennial-alias/js/common/gitPull.js' );
+const npmUpdate = require( '../../../perennial-alias/js/common/npmUpdate.js' );
 
 const REPO_DEPENDENCIES = [ 'chipper', 'perennial', 'perennial-alias' ];
 
Index: src/server/translationApi/requestBuilds.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/server/translationApi/requestBuilds.js b/src/server/translationApi/requestBuilds.js
--- a/src/server/translationApi/requestBuilds.js	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/src/server/translationApi/requestBuilds.js	(date 1734467022393)
@@ -8,8 +8,8 @@
  */
 
 import axios from 'axios';
-import simPhetioMetadata from '../../../../perennial/js/common/simPhetioMetadata.js';
-import SimVersion from '../../../../perennial/js/browser-and-node/SimVersion.js';
+import simPhetioMetadata from '../../../../perennial-alias/js/common/simPhetioMetadata.js';
+import SimVersion from '../../../../perennial-alias/js/browser-and-node/SimVersion.js';
 import privateConfig from '../../common/privateConfig.js';
 import getDependencies from './getDependencies.js';
 import getSimVersionObject from './getSimVersionObject.js';
Index: scripts/js/patch-code-point-combinations.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/js/patch-code-point-combinations.js b/scripts/js/patch-code-point-combinations.js
--- a/scripts/js/patch-code-point-combinations.js	(revision 79fe6d34a8f7028c272142f11cb8c410fe625c70)
+++ b/scripts/js/patch-code-point-combinations.js	(date 1734467022375)
@@ -10,7 +10,7 @@
 
 // imports
 const fs = require( 'fs' );
-const getActiveRepos = require( '../../../perennial/js/common/getActiveRepos' );
+const getActiveRepos = require( '../../../perennial-alias/js/common/getActiveRepos' );
 
 const activeRepos = getActiveRepos();
